src/librustc_mir/interpret/eval_context.rs

```diff
@@ -777,8 +780,13 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Our result will later be validated anyway, and there seems no good reason
         // to have to fail early here.  This is also more consistent with
         // `Memory::get_static_alloc` which has to use `const_eval_raw` to avoid cycles.
-        let val = self.tcx.const_eval_raw(param_env.and(gid))?;
-        self.raw_const_to_mplace(val)
+        let raw = self.tcx.const_eval_raw(param_env.and(gid))?;
+        // For promoteds and constants ZSTs don't need a guaranteed address
```

Comment:
Don't *need* is one thing, but why would it be *wrong*?

Reply:
It's working, it's just that we have tests checkin whether the value of a `&()` is `1` and I wanted to keep these tests working.

Reply:
This PR does not change anything here anymore and the guarantee comment is gone